backend/monitoring/dashboard.py

```
@@ -707,6 +732,343 @@ async def trigger_test(test_type: str, admin=Depends(get_current_admin_dependenc
+        logger.error(f"Failed to get test suites: {e}")
+        return StandardResponse(
+            status="error",
+            message=f"Failed to get test suites: {str(e)}",
+            data=[]
+        )
+
+@router.get("/business-logic-validation")
+async def get_business_logic_validation_status(admin=Depends(get_current_admin_dependency)):
+    """Get business logic validation status and recent results"""
+    try:
+        conn = await asyncpg.connect(DATABASE_URL)
```

Ruff (0.12.2)

Undefined name `asyncpg`

(F821)